types/message.go

```diff
@@ -107,3 +107,34 @@ func (msg *Message) Equals(other *Message) bool {
 		msg.Method == other.Method &&
 		bytes.Equal(msg.Params, other.Params)
 }
+
+// MessageCollection tracks a group of messages and assigns it a cid.
+type MessageCollection []*SignedMessage
+
+// TODO #3078 the panics here and in types.Block should go away.  We need to
+// keep them in order to use the ipld cborstore with the default hash function
+// because we need to implement hamt.cidProvider which doesn't handle errors.
+// We can clean all this up when we can use our own CborIpldStore with the hamt.
+
+// Cid returns the cid of the message collection.
+func (mC MessageCollection) Cid() cid.Cid {
+	nd, err := cbor.WrapObject(mC, DefaultHashFunction, -1)
```

Comment:
You could shorten this code by having them each call one private free function with themselves as a parameter.